trio/_core/_run.py

```diff
@@ -1557,6 +1565,12 @@ def run_impl(runner, async_fn, args):
         # change too, like the deadlines tie-breaker and the non-deterministic
         # ordering of task._notify_queues.)
         batch = list(runner.runq)
+        if _ALLOW_DETERMINISTIC_SCHEDULING:
```

Comment:
I'm neither a native speaker or a fluent speaker, but I'm confused by `_ALLOW_DETERMINISTIC_SCHEDULING`. To me, that means "the scheduling could be deterministic, because it's allowed, but maybe it's not? who knows".

Is `_ENFORCE_DETERMINISTIC_SCHEDULING` a better name?

Reply:
> To me, [`_ALLOW_DETERMINISTIC_SCHEDULING`] means "the scheduling could be deterministic, because it's allowed, but maybe it's not? who knows".

That's exactly what it *does* mean!  If this flag is set to True, you *can* enable determinism by seeding the Random instance `_r`, but it's still not deterministic by default.